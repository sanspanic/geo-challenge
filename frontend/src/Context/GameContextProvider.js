import React, { useState, useEffect } from "react";
import GameContext from "./GameContext";
import geoApi from "../API/geoAPI";

const GameContextProvider = ({ children }) => {
  const [speedBonus, setSpeedBonus] = useState(0);
  const [width, setWidth] = useState(100);
  const [level, setLevel] = useState(1);
  const [score, setScore] = useState(0);
  const [countries, setCountries] = useState([]);
  const [rank, setRank] = useState(1);
  const [hasLoaded, setHasLoaded] = useState(false);
  const [lives, setLives] = useState(5);
  const [finalScore, setFinalScore] = useState(0);

  const [status, setStatus] = useState({
    isActive: false,
    isLost: true,
    isWon: false,
  });

  useEffect(() => {
    const getCountryData = async () => {
      try {
        const res = await geoApi.getData();
        setCountries(res);
        setHasLoaded(true);
      } catch (err) {
        console.log(err);
      }
    };
    getCountryData();
  }, []);

  return (
    <GameContext.Provider
      value={{
        level,
        setLevel,
        score,
        setScore,
        countries,
        setCountries,
        hasLoaded,
        status,
        setStatus,
        lives,
        setLives,
        width,
        setWidth,
        speedBonus,
        setSpeedBonus,
        rank,
        setRank,
        finalScore,
        setFinalScore,
      }}
    >
      {children}
    </GameContext.Provider>
  );
};

export default GameContextProvider;
