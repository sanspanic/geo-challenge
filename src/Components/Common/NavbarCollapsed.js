import React, { useContext } from "react";
import { Link } from "react-router-dom";
import Planet from "../../Assets/planet.png";
import AuthContext from "../../Context/AuthContext";
import Button from "../Common/Button";
import LogoutField from "./LogoutField";

const NavbarCollapsed = ({ setIsMenuOpen }) => {
  const { user } = useContext(AuthContext);
  return (
    <div className="absolute top-0 left-0 w-full">
      <div className="p-5 bg-black  rounded shadow-sm">
        <div className="flex items-center justify-between mb-4">
          <div>
            <Link
              href="/"
              aria-label="Geo Challenge"
              title="Geo Challenge"
              className="inline-flex items-center"
            >
              <img width={30} src={Planet}></img>
              <span className="ml-5 text-xl font-bold tracking-wide text-gray-800 uppercase text-gradient-purple">
                Geo Challenge
              </span>
            </Link>
          </div>
          <div>
            <button
              aria-label="Close Menu"
              title="Close Menu"
              className="p-2 -mt-2 -mr-2 transition duration-200 rounded hover:bg-gray-200 focus:bg-gray-200 focus:outline-none focus:shadow-outline"
              onClick={() => setIsMenuOpen(false)}
            >
              <svg className="w-5 text-gray-600" viewBox="0 0 24 24">
                <path
                  fill="currentColor"
                  d="M19.7,4.3c-0.4-0.4-1-0.4-1.4,0L12,10.6L5.7,4.3c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l6.3,6.3l-6.3,6.3 c-0.4,0.4-0.4,1,0,1.4C4.5,19.9,4.7,20,5,20s0.5-0.1,0.7-0.3l6.3-6.3l6.3,6.3c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3 c0.4-0.4,0.4-1,0-1.4L13.4,12l6.3-6.3C20.1,5.3,20.1,4.7,19.7,4.3z"
                />
              </svg>
            </button>
          </div>
        </div>
        <nav>
          <ul className="space-y-4">
            {user.username ? (
              <>
                <li>
                  <Link
                    href="/game"
                    aria-label="Product pricing"
                    title="Product pricing"
                    className="font-medium tracking-wide text-earthblue-400 transition-colors duration-200 hover:text-earthgreen-400"
                  >
                    Game
                  </Link>
                </li>
                <li>
                  <Link
                    to="/profile"
                    aria-label="profile"
                    title="profile"
                    className="font-medium tracking-wide text-earthblue-400 transition-colors duration-200 hover:text-earthgreen-400"
                  >
                    Profile
                  </Link>
                </li>
                <li>
                  <LogoutField />
                </li>
              </>
            ) : (
              <>
                <li>
                  <a
                    href="/register"
                    aria-label="Product pricing"
                    title="Product pricing"
                    className="font-medium tracking-wide text-earthblue-400 transition-colors duration-200 hover:text-earthgreen-400"
                  >
                    Register
                  </a>
                </li>
                <li>
                  <a
                    href="/login"
                    aria-label="About us"
                    title="About us"
                    className="font-medium tracking-wide text-earthblue-400 transition-colors duration-200 hover:text-earthgreen-400"
                  >
                    Login
                  </a>
                </li>
              </>
            )}

            <li>
              <Button gradient="purple" text="leaderboard" />
            </li>
          </ul>
        </nav>
      </div>
    </div>
  );
};

export default NavbarCollapsed;
