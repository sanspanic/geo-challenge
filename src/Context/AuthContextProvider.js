import React, { useState } from "react";
import AuthContext from "./AuthContext";

const AuthContextProvider = ({ children }) => {
  const [user, setUser] = useState(() => {
    const user = window.localStorage.getItem("user");
    return user !== null ? JSON.parse(user) : {};
  });

  return (
    <AuthContext.Provider value={{ user, setUser }}>
      {children}
    </AuthContext.Provider>
  );
};

export default AuthContextProvider;
