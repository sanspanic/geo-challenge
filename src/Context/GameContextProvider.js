import React, { useState } from "react";
import GameContext from "./GameContext";

const GameContextProvider = ({ children }) => {
  const [level, setLevel] = useState({ num: 1, title: "Guess The Flag" });
  const [score, setScore] = useState(0);

  return (
    <GameContext.Provider value={{ level, setLevel, score, setScore }}>
      {children}
    </GameContext.Provider>
  );
};

export default GameContextProvider;
