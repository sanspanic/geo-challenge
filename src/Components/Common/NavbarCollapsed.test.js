import React from "react";
import { render } from "@testing-library/react";
import NavbarCollapsed from "./NavbarCollapsed";
import GameContextProvider from "../../Context/GameContextProvider";
import AuthContextProvider from "../../Context/AuthContextProvider";
import { MemoryRouter } from "react-router-dom";

test("renders without crashing", () => {
  render(
    <AuthContextProvider>
      <GameContextProvider>
        <MemoryRouter>
          <NavbarCollapsed />
        </MemoryRouter>
      </GameContextProvider>
    </AuthContextProvider>
  );
});

test("matches snapshot", () => {
  const { asFragment } = render(
    <AuthContextProvider>
      <GameContextProvider>
        <MemoryRouter>
          <NavbarCollapsed />
        </MemoryRouter>
      </GameContextProvider>
    </AuthContextProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
