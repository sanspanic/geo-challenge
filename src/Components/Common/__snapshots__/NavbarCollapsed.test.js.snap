// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="absolute top-0 left-0 w-full"
  >
    <div
      class="p-5 bg-black  rounded shadow-sm"
    >
      <div
        class="flex items-center justify-between mb-4"
      >
        <div>
          <a
            aria-label="Geo Challenge"
            class="inline-flex items-center"
            href="/"
            title="Geo Challenge"
          >
            <img
              alt="planet"
              src="planet.png"
              width="30"
            />
            <span
              class="ml-5 text-xl font-bold tracking-wide text-gray-800 uppercase text-gradient-purple"
            >
              Geo Challenge
            </span>
          </a>
        </div>
        <div>
          <button
            aria-label="Close Menu"
            class="p-2 -mt-2 -mr-2 transition duration-200 rounded hover:bg-gray-200 focus:bg-gray-200 focus:outline-none focus:shadow-outline"
            title="Close Menu"
          >
            <svg
              class="w-5 text-gray-600"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.7,4.3c-0.4-0.4-1-0.4-1.4,0L12,10.6L5.7,4.3c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l6.3,6.3l-6.3,6.3 c-0.4,0.4-0.4,1,0,1.4C4.5,19.9,4.7,20,5,20s0.5-0.1,0.7-0.3l6.3-6.3l6.3,6.3c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3 c0.4-0.4,0.4-1,0-1.4L13.4,12l6.3-6.3C20.1,5.3,20.1,4.7,19.7,4.3z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <nav>
        <ul
          class="space-y-4"
        >
          <li>
            <a
              aria-label="Product pricing"
              class="font-medium tracking-wide text-earthblue-400 transition-colors duration-200 hover:text-earthgreen-400"
              href="/register"
              title="Product pricing"
            >
              Register
            </a>
          </li>
          <li>
            <a
              aria-label="About us"
              class="font-medium tracking-wide text-earthblue-400 transition-colors duration-200 hover:text-earthgreen-400"
              href="/login"
              title="About us"
            >
              Login
            </a>
          </li>
          <li>
            <a
              class="self-center"
              href="/leaderboard"
            >
              <button
                class="mx-2 px-4 py-2 bg-gradient-purple rounded font-bold text-white transition duration-400 hover:text-black transform-gpu hover:scale-110"
              >
                Leaderboard
              </button>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
